src/components/DocumentPage.js

```diff
@@ -0,0 +1,116 @@
+import { request } from "../api/api.js";
+import DocumentList from "./DocumentList.js";
+import { push } from "../routes/router.js";
+import { setItem, removeItem } from "../utils/storage.js";
+import LinkButton from "./LinkButton.js";
+
+export default function DocumentPage({ $target, initialState }) {
+  const $documentPage = document.createElement("div");
+
+  $documentPage.className = "DocumentPage";
+
+  $documentPage.style = `
+    width:30%;
+    height:100vh;
+    background-color:#fffafa	;
+  `;
+
+  this.state = initialState;
+
+  this.setState = async (nextState) => {
```

Comment:
```suggestion
  this.setState = (nextState) => {
```
`await`을 할 명령이 없는 것 같아 `async`를 제거하는게 좋을 것 같습니다!!